File: 11.Stack/Python/14.SmartTeacher.py
def process_string(s):
    stack = []
    for char in s:
        if char == '#':
            if stack:
                stack.pop()
        else:
            stack.append(char)
    return ''.join(stack)

def smart_teacher():
    answer = "ab#c"
    solution = "ad#c"

    final_answer = process_string(answer)
    final_solution = process_string(solution)

    if final_answer == final_solution:
        print("CORRECT")
    else:
        print("WRONG")

# Run the function
smart_teacher()

File: 11.Stack/Javascript/14.SmartTeacher.js
function processString(s) {
    const stack = [];
    for (const char of s) {
        if (char === '#') {
            if (stack.length > 0) {
                stack.pop();
            }
        } else {
            stack.push(char);
        }
    }
    return stack.join('');
}


const answer = "ab#c";
const solution = "ad#c";

const finalAnswer = processString(answer);
const finalSolution = processString(solution);

if (finalAnswer === finalSolution) {
    console.log("CORRECT");
} else {
    console.log("WRONG");
}

